src/widgets/auto-styler.js

```diff
@@ -0,0 +1,141 @@
+var cdb = require('cartodb.js');
+var CategoryColors = require('./category/category-colors');
+var AutoStyler = cdb.core.Model.extend({
+  initialize: function (dataviewModel) {
+    this.dataviewModel = dataviewModel;
+    this.colors = new CategoryColors();
+    this.vector = !!this.dataviewModel._dataProvider;
+  },
+
+  getStyle: function () {
+    var style;
+    var widgetType = this.dataviewModel.get('type');
+    if (widgetType === 'category') {
```

Comment:
Just worth mentioning that this could be a good opportunity to [replace this conditional with polymorphism](https://sourcemaking.com/refactoring/replace-conditional-with-polymorphism) (in combination with a factory):

```
var AutoStylerFactory = {
  get: function (dataviewModel) {
    if (dataviewModel.get('type') === 'category') {
      return new CategoryAutoStyler(dataviewModel);
    } else if (dataviewModel.get('type') === 'histogram') {
      return new HistogramAutoStyler(dataviewModel);
    } else {
      throw new Error('dataview type not supported');
    }
  }
}
...
var histogramAutoStyler = AutoStylerFactory.get(histogramDataview);
histogramAutoStyler.getStyle();
...
var CategoryAutoStyler = AutoStylerFactory.get(categoryDataview);
categoryAutoStyler.getStyle();
```

This way widgets could something like:
```
this.autoStyler = AutoStylerFactory.get(this.dataviewModel);
...
this.autoStyler.getStyle();
```